pkg/resource/resource.go

```diff
@@ -119,3 +119,8 @@ func (res *Resource) Resources() []Resource {
 
 	return []Resource{*res}
 }
+
+// QualifiedName returns a string for a signature in the format version/kind/namespace/name
+func (sig *Signature) QualifiedName() string {
+	return fmt.Sprintf("%s/%s/%s/%s", sig.Version, sig.Kind, sig.Namespace, sig.Name)
```

Comment:
do you think it would make sense to default the namespace to default here? I was wondering about the double // in your test :thinking: Generally I'm wondering if there might be a "standardized" format to refer to a resource like this.

Reply:
> do you think it would make sense to default the namespace to default here? I was wondering about the double // in your test thinking

`kubeval` has this option:

```
-n, --default-namespace string              Namespace to assume in resources if no namespace is set in metadata:namespace (default "default")
```

`kubectl diff` as we can see in the link below leaves it as an empty string.

The main issue I can think of if we set a default is for cluster level resources like `ClusterRole` or `Namespace` which will risk to have a "default" namespace in the output

> Generally I'm wondering if there might be a "standardized" format to refer to a resource like this.

`kubectl diff` uses a similar format for diffed files, expect with dots as separator, likely because filenames cannot contain slashes:
https://github.com/kubernetes/kubectl/blob/v0.23.1/pkg/cmd/diff/diff.go#L412-L418

Group and Version are already separated by a slash. And when using `kubectl`, resources can be referenced as `<kind>/<name>`. Although I do not remember observing a standard format anywhere, concatenating everything with slashes in this order sort of makes sense to me. I am open to change if you think of anything

Reply:
Thanks for the clarification :1st_place_medal: